src/classes/BDI_MigrationMappingHelper.cls

```diff
@@ -123,8 +123,8 @@ public class BDI_MigrationMappingHelper {
                 if (fieldNameIsUsable && objectNameIsUsable && objectAPINameIsUsable) {
```

Comment:
All of the `isUsable` checks can be done inside the `HelpTextFieldMapping` inner class.  So you should be able to instantiate the inner class by passing in the sourceField DFR after line 110.  Then you can call 
`if(helpTextFieldMapping.isUsable()` on line 114.  If it is usable, instantiate the `DataImportFieldMapping` instance.  If not, add the `HelpTextFieldMapping` to your list `badHelpTextFieldDescribeResults`.  Let me know what you think!